index.src.html

```diff
@@ -1229,6 +1242,74 @@ <h2 id="sample-reports">Sample Reports</h2>
   </div>
 </section>
 
+<section>
+
+</section>
+  <h2 id="automation">Automation</h2>
+
+  For the purposes of user-agent automation and application testing, this
+  document defines a number of <a>extension commands</a> for the [[WebDriver]]
+  specification.
+
+  <h3 id="generate-test-report-command">Generate Test Report</h3>
+
+  The <dfn>Generate Test Report</dfn> <a>extension command</a> simulates the
+  generation of a <a>report</a> for the purposes of testing. This report will be
+  routed through any defined reporting <a>endpoints</a> and observed by
+  any <a>registered</a> <a>reporting observers</a>, as per normal.
+
+  The <a>extension command</a> is defined as follows:
+
+  <table>
+    <tbody>
+      <tr>
+        <th>HTTP Method</th>
+        <th><a lt="extension command prefix">Prefix</a></th>
+        <th><a lt="extension command name">Name</a></th>
+      </tr>
+      <tr>
+        <td>POST</td>
+        <td>/session/{session id}/reporting</td>
+        <td>generate_test_report</td>
+      </tr>
+    </tbody>
+  </table>
+
+  <pre class='idl'>
+    dictionary GenerateTestReportParameters {
+      required DOMString message;
+      DOMString group;
+    };
+  </pre>
+
+  The <a>remote end steps</a> are:
+
+  1. If |parameters| is not a JSON <a>Object</a>, return a <a>WebDriver
+     error</a> with <a>WebDriver error code</a> <a>invalid argument</a>.
+
+  2. Let |message| be the result of <a>trying</a> to get |parameters|'s
+     {{GenerateTestReportParameters/message}} property.
+
+  3. If |message| is null, return a <a>WebDriver error</a> with <a>WebDriver
+     error code</a> <a>invalid argument</a>.
+
+  4. Let |group| be the result of <a>trying</a> to get |parameters|'s
+     {{GenerateTestReportParameters/group}} property.
+
+  5. If |group| is null, set |group| to "default".
+
+  6. Let |body| be a new object that can be serialized into a <a>JSON text</a>,
+     containing a single string field, |body_message|.
+
+  7. Set |body_message| to |message|.
+
+  8. Let |settings| be the <a>environment settings object</a> of the
+     <a>current browsing context</a>'s <a>active document</a>.
+
+  9. Execute [[#queue-report]] with |body|, "test", |group|, and |settings|.
```

Comment:
What happens if this fails/throws an error?

Reply:
I think we're okay here.  The `#queue-report` algorithm always succeeds — it just adds the report to a queue and immediately returns.

Reply:
[Though that does mean we might need another extension command that lets you trigger the `#send-reports` algorithm, and *that* one does have interesting failure modes]

Reply:
What if there is some unforeseen memory error? Or it's implemented in some more complicated way in a different browser?

Reply:
I'd argue that there's not much that you can do in that case.  Report delivery is already best-effort, not guaranteed (see §1.1).  A report can be dropped at any point down the line (by being evicted when a later report is queued, for instance, or during a GC run), and there's no way to signal that back to whatever originally generated the report.

Reply:
Ahh I see what you are saying now. In the scope of just the reporting API that makes sense to me, but I think I would still prefer to have some sort of catch for an error. Maybe WebDriver is doing this by invoking a call to another system and that system changes the name of the method WebDriver usually calls. In that case we never reach the queue but we still error out. I think Unknown Error in the WebDriver spec could be used for this case.

WYT?

Reply:
Can you give an example of a specific error condition that you'd want to write a test for?

Reply:
If ChromeDriver implements this it will probably use the DevTools Protocol to dispatch the queue-report. What if there is an error in the in protocol layer?

Reply:
After a conversation offline with Paul I've changed my opinion and I think this is good to go as is. For further reference this can be compared to step 6 in the maximize window command in the WebDriver (spec)[https://w3c.github.io/webdriver/#maximize-window] which is handled in a similar way to this.